build/builder.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# Determine the directory of the script
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+
+set -o allexport
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR/build.env"
+
+# Find and source all build.env files in the $SCRIPT_DIR/build/*/ directories
```

Comment:
Did you consider using docker compose `.env` file support? This could spare us this extra shell code here...


```
# .env
ENV_VAR=myvalue
```

```
# docker-compose.yml
services:
  build:
    build:
      context: .
      args:
        - BUILD_ENV_VAR=${ENV_VAR}
```

```
# Dockerfile
FROM debian
ARG BUILD_ENV_VAR
RUN echo "Using env var value: ${BUILD_ENV_VAR}"
```

Running:
```
❱ docker compose build     
[...]
=> CACHED [build 2/2] RUN echo "Using env var value: myvalue"
[...]
```

Reply:
Ah yes, tried many things for that pre-eminently the `.env` file but also `services.<service_name>.env_file`, but none of them work to get env vars into build / Build Args in compose. I think that you'll see that it turns-out to be not be particularly important as you see the subsequent steps  as we once again move the parallel building outside of anything docker and use the nx task runner.

Reply:
This branch is an entirely functional improvement of the current state of things, but also very much a rest stop on the way to the later work.